Tests/BackendIntegrationTests/OfflineStoreKitIntegrationTests.swift

```diff
@@ -246,6 +246,25 @@ class OfflineStoreKit1IntegrationTests: BaseOfflineStoreKitIntegrationTests {
         )
     }
 
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func testCallToGetCustomerInfoWithPendingTransactionsPostsReceiptOnlyOnce() async throws {
+        self.serverDown()
+
+        _ = try await self.purchaseMonthlyProduct()
+        _ = try await Purchases.shared.purchase(product: self.product(Self.group3MonthlyTrialProductID))
+
+        self.serverUp()
+
+        let customerInfo = try await self.purchases.customerInfo(fetchPolicy: .fetchCurrent)
+        expect(customerInfo.activeSubscriptions).to(haveCount(2))
+
+        self.logger.verifyMessageWasLogged(
```

Comment:
This test is failing right now because we end up with 2 cache keys:
```
PostReceiptDataOperation $RCAnonymousID:2e0dc9480ff946ff99d016f1e025aeb3-true-42096aeb16be0c361f0a6b310786c3cdac44d80edbd55d0042d02bf6cbae048c
-com.revenuecat.monthly_4.99.1_week_intro-4.99-USD-USA-2-0-7096FF06-P1M-P1W-2-com.revenuecat.monthly_4.99.1_free_week
--false
```
```
PostReceiptDataOperation $RCAnonymousID:2e0dc9480ff946ff99d016f1e025aeb3-true-f7c4c48b673a12d6d916de52fd99eff7bce705870ee67bbf4a4030b6775dfcec
-com.revenuecat.monthly.1.99.1_free_week-1.99-USD-USA-2-0-45445385-P1M-P1W-2
--false
```